infrastructure/index.ts

```diff
@@ -50,43 +52,28 @@ const contentBucket = new aws.s3.Bucket(
     },
 );
 
-// contentBucket needs to have the "public-read" ACL so its contents can be ready by CloudFront and
-// served. But we deny the s3:ListBucket permission to prevent unintended disclosure of the bucket's
-// contents.
-const denyListPolicyState: aws.s3.BucketPolicyArgs = {
-    bucket: contentBucket.bucket,
-    policy: contentBucket.arn.apply((arn: string) => JSON.stringify({
+// The website bucket needs to have the "public-read" ACL so its contents can be read by
+// CloudFront and served. But we deny the s3:ListBucket permission to anyone but account
+// users to prevent unintended disclosure of the bucket's contents.
+const policy = new aws.s3.BucketPolicy("website-bucket-policy", {
+    bucket: websiteBucket.bucket,
+    policy: websiteBucket.arn.apply(arn => JSON.stringify({
         Version: "2008-10-17",
         Statement: [
             {
                 Effect: "Deny",
                 Principal: "*",
                 Action: "s3:ListBucket",
                 Resource: arn,
+                Condition: {
+                    StringNotEquals: {
+                        "aws:PrincipalAccount": config.principalAccount,
+                    },
+                },
             },
         ],
     })),
-};
-const denyListPolicy = new aws.s3.BucketPolicy("deny-list", denyListPolicyState);
-
-// websiteBucket stores the static content to be served via the CDN.
-const websiteBucket = new aws.s3.Bucket(
```

Comment:
Just confirming: The `websiteBucket` resource is still around. It's just the `contentBucket` that was removed, because we now push the data directly into `websiteBucket`. (Since we switched over to the new setup, right?)

Reply:
That’s right — the diff just looks terrifying, presumably because of the way I duplicated `contentBucket`’s declaration during the cutover.